IHP/IDE/Defaults/TableColumnDefaults.hs

```diff
@@ -0,0 +1,425 @@
+{-|
+Module: Test.IDE.Defaults.TableColumnDefaults
+Copyright: (c) digitally induced GmbH, 2020
+-}
+module IHP.IDE.Defaults.TableColumnDefaults where
+
+import Test.Hspec
+import IHP.Prelude
+import qualified IHP.IDE.CodeGen.ControllerGenerator as ControllerGenerator
+import IHP.ViewPrelude (cs, plain)
+import qualified Text.Megaparsec as Megaparsec
+import IHP.IDE.CodeGen.Types
+import IHP.IDE.SchemaDesigner.Types
+import IHP.NameSupport
+import IHP.IDE.SchemaDesigner.Types
+
+
+{- | Takes a `Text` value for the name and creates a default column where all values are empty lists, 
+'Nothing' or 'False'; and for 'ColumnType', the default is 'PUUID'.
+
+Add a new field to the 'Column' type in the 
+"IHP.IDE.SchemaDesigner.Types" file and then 
+set its default value here
+
+Defined as:
+
+@
+defColumn :: Column
+defColumn = Column { 
+                 name = ""
+               , columnType = PUUID 
+               , defaultValue = Nothing
+               , notNull = False 
+               , isUnique = False
+               , generator = Nothing
+               }
+@
+
+If you want a different 'PostgresType' you will need to 
+specify with like so by either using a function:
+
+@setColumnType pgt = defColumn {columnType = pgt}@
+
+Or
+
+Just as part of where you're calling it:
+
+@someDefaultColumnType = defColumn {columnType = PDate}@
+-}
+defColumn :: Column
+defColumn = Column { 
+                 name = ""
+               , columnType = PUUID 
+               , defaultValue = Nothing
+               , notNull = False 
+               , isUnique = False
+               , generator = Nothing
+               }
+
+
+{- | Creates a table where all values are empty lists, including columns. @unlogged@ is set to 'False'.
+
+Defined as such:
+
+@
+defCreateTable :: CreateTable
+defCreateTable = CreateTable {
+                        name = ""
+                        , columns = []
+                        , primaryKeyConstraint = PrimaryKeyConstraint []
+                        , constraints = []
+                        , unlogged = False
+                        }
+@
+
+-}
+defCreateTable :: Text -> CreateTable
+defCreateTable t = CreateTable {
+                        name = t
+                        , columns = []
+                        , primaryKeyConstraint = PrimaryKeyConstraint []
+                        , constraints = []
+                        , unlogged = False
+                        }
+
+-- | Takes a name for our table and a list of column and inserts the list
+--  into to our default table.
+defCreateTableWCol :: Text -> [Column] -> CreateTable
```

Comment:
IMO this can be just done via `defCreateTable`. When you have a table that has no columns, you could still pass an empty list like `defCreateTable "myTableWithoutCols" []`.

Reply:
Yes, I agree now, especially now I see how often each pattern appears, it makes more sense to define this in these ways - will alter :)